src/buildings/building.gd

```diff
@@ -39,14 +39,35 @@ func _ready():
 	set_original_color()
 	build_timer_ui.visible = false
 	TickManager.tick.connect(_on_tick)
-	ResourceManager.upgrade_acquired.connect(apply_upgrades)
-	apply_upgrades()
+	ResourceManager.upgrade_acquired.connect(check_for_adjacency_multiplier)
+	check_for_adjacency_multiplier()
 
 	if placed and building_complete:
 		_setup_scan_for_nearby_bonus()
 
+func _process(delta):
+	# TODO: Optimize this
+	if Input.is_action_just_pressed("cancel_place_building"):
+		if is_hover and can_delete and not preview:
+			if is_constructing and not is_deconstructing:
+				cancel_construction()
+			elif building_complete and is_deconstructing:
+				cancel_deconstruction()
+			else:
+				set_building_remove()
+	if not placed and preview:
+		if collider.has_overlapping_areas() or collider.has_overlapping_bodies() or outside_gridmap:
+			color_sprite(1, 0, 0, 0.5)
+			placeable = false
+		else:
+			color_sprite(0, 1, 0, 0.5)
+			placeable = true
+
+func check_for_adjacency_multiplier(_unused_var = null):
```

Comment:
Do we need to pass this `_unused_var` parameter to this function?

Reply:
yeah, since some of the signal connect to this function has parameter, if it not provided an unused var, godot complain about it